taichi/transforms/constant_fold.cpp

```diff
@@ -44,28 +116,49 @@ class ConstantFold : public BasicStmtVisitor {
     }
   }
 
+  static bool jit_from_binary_op(TypedConstant &tc, BinaryOpType op,
+      const TypedConstant &lhs, const TypedConstant &rhs)
+  {
+    if (lhs.dt != DataType::i32 || rhs.dt != DataType::i32 || tc.dt != DataType::i32)
+      return false;
+    auto kernel_name = fmt::format("jit_constexpr_{}", 0);
+    auto func = [&] () {
+      auto lhstmt =
+        Stmt::make<ConstStmt>(LaneAttribute<TypedConstant>(lhs));
```

Comment:
Let's use `ArgLoadStmt` for caching the JIT kernels.

If we compile a kernel for each statement that can be constant folded, it will be very time-consuming --- we'll have to compile O(n) kernels (where n is the number of statements).

On the other hand, if we cache the kernels for each combination of types, the number of kernels we need to compile will be a constant. This will reduce compile time, especially for large programs.